godot/RandomWalker/Actors/Player.gd

```diff
@@ -3,43 +3,33 @@ extends "Actor.gd"
 
 const TIME_DEATH := 0.2
 
-onready var danger_detector: Area2D = $DangerDetector
-onready var tween: Tween = $Tween
+@onready var danger_detector: Area2D = $DangerDetector
 
 
 func _on_DangerDetector_body_entered(_body: Node):
 	set_physics_process(false)
 	
-	tween.interpolate_property(self, "scale", null, Vector2.ZERO, TIME_DEATH, Tween.TRANS_LINEAR)
-	tween.start()
-	
-	yield(tween, "tween_all_completed")
-	queue_free()
-
+	var tween = get_tree().create_tween().bind_node(self)
```

Comment:
Same idea with tween using the node method `var tween := create_tween()`. Also when declaring variables always use typed walrus assignment: `var tween :=`